app/views/main/index.erb

```diff
@@ -8,12 +8,14 @@
 <% @challenges.each do |c| %>
     <div>
         <h5 class="challenge">
-            <%= link_to c.title, challenge_path(c) %> - <%= @entry_counts[c.id] || 0 %> entries
+          <%= link_to c['title'], challenge_path(c['_id']) %> - <%= c['count_entries'] %> entries
         </h5>
-        <p><%= c.description %></p>
+        <p><%= c['description'] %></p>
```

Comment:
Curious, any particular reason why you switched to `[]` syntax?

Reply:
All the Model.collection.aggregate return a big hash. It is not mapped to our model because we can add any new fields.

Some month ago I was against that but after playing with other langage I can't see the problem.

But if you are against, and prefer the dot syntax for any reason, I can instanciate a 'Struct' (or any custom object) or maybe try with Challenge.

My preference are 
- keep this hash, (and document method as you ask)
- create a custom Struct tailored for this query
- instanciate a challenge, but I need more time to understand how it behaves with new fields

Reply:
All the Model.collection.aggregate return a big hash. It is not mapped to our model because we can add any new fields.

Some month ago I was against that but after playing with other langage I can't see the problem.

But if you are against, and prefer the dot syntax for any reason, I can instanciate a 'Struct' (or any custom object) or maybe try with Challenge.

My preference are 
- keep this hash, (and document method as you ask)
- create a custom Struct tailored for this query
- instanciate a challenge, but I need more time to understand how it behaves with new fields

Reply:
Hash is fine, thanks for the context.